pgmpy/inference/ExactInference.py

```diff
@@ -228,6 +230,45 @@ def induced_graph(self, elimination_order):
         >>> inference.induced_graph(['C', 'D', 'A', 'B', 'E'])
         <networkx.classes.graph.Graph at 0x7f34ac8c5160>
         """
+        eliminated_variables = set()
+        working_factors = {node: {factor for factor in self.factors[node]}
+                           for node in self.factors}
+
+        # The set of cliques should be in the induced graph
+        cliques = set()
+        for factors in working_factors.values():
+            for factor in factors:
+                cliques.add(tuple(factor.scope()))
+
+        # If the elimination_order list is incomplete, raise an error
+        if len(elimination_order) < len(self.variables):
+            raise ValueError("Elimination order incomplete")
+
+        # Removing all the factors containing the variables which are
+        # eliminated (as all the factors should be considered only once)
+        for var in elimination_order:
+            factors = [factor for factor in working_factors[var]
+                       if not set(factor.variables).intersection(eliminated_variables)]
+            phi = factor_product(*factors)
```

Comment:
@pratyakshs Do we really need to actually multiply the factors? We will not be needing the actual values anywhere and just be needing the scopes. So, I think this should be avoidable.